Rank/Obj_Rank_FinalInd_calc.php

```
@@ -278,6 +290,23 @@ protected function calcFromPhase($event, $realphase)
+								? (strlen(preg_replace("/\d/", "", $myRow->SetPoints)) + 1) * $myRow->DiEndArrows
+								: ($myRow->DiArrows ?: 1))), 3);
+						$avgTie = round(valutaArrowString($myRow->Tiebreak)
+							/ (strlen(trim($myRow->Tiebreak)) ?: 1), 3);
+						$oppAvgMatch = round($myRow->OppScore / (strlen(trim($myRow->OppArrowstring))
+							?: (strlen(preg_replace("/\d/", "", $myRow->OppSetPoints))
+								? (strlen(preg_replace("/\d/", "", $myRow->OppSetPoints)) + 1) * $myRow->DiEndArrows
+								: ($myRow->DiArrows ?: 1))), 3);
+						$oppAvgTie = round(valutaArrowString($myRow->OppTiebreak)
+							/ (strlen(trim($myRow->OppTiebreak)) ?: 1), 3);
+
+						safe_w_sql("UPDATE Finals SET FinAverageMatch='{$avgMatch}', FinAverageTie='{$avgTie}' WHERE FinTournament={$this->tournament} AND FinEvent='{$EventToUse}' AND FinMatchNo='{$myRow->RealMatchNo}'");
```

PHPStan (2.1.46)

[warning] Access to an undefined property Obj_Rank_FinalInd_calc::$tournament.
Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property

(property.notFound)